python/ray/air/util/data_batch_conversion.py

```diff
@@ -238,5 +238,5 @@ def _cast_tensor_columns_to_ndarrays(df: pd.DataFrame) -> pd.DataFrame:
         # Try to convert any tensor extension columns to ndarray columns.
         for col_name, col in df.items():
             if isinstance(col.dtype, TensorDtype):
-                df.loc[:, col_name] = pd.Series(list(col.to_numpy()))
+                df[col_name] = pd.Series(list(col.to_numpy()))
```

Comment:
Same here, we want to mutate (overwrite) the column in-place, so `df.loc[:, col_name] = pd.Series(list(col.to_numpy()))` seems correct to me.